pkg/manager/manager.go

```diff
@@ -488,16 +525,27 @@ func (o Options) AndFrom(loader config.ControllerManagerConfiguration) (Options,
 		o.LivenessEndpointName = newObj.Health.LivenessEndpointName
 	}
 
-	if o.Port == 0 && newObj.Webhook.Port != nil {
-		o.Port = *newObj.Webhook.Port
+	webhookServer := &webhook.Server{}
+	if newObj.Webhook.Port != nil {
+		if o.Port == 0 {
+			o.Port = *newObj.Webhook.Port
+		}
+		webhookServer.Port = *newObj.Webhook.Port
```

Comment:
I'm probably missing something. Shouldn't webhookServer.Port be o.Port if o.Port is != 0?

Reply:
Hm or is the idea that .Webhook.Port has the higher priority? It's just a bit confusing as in all the other places above we only set anything on o Options if the field there is not set

Reply:
`webhookServer.Port` is set regardless because it's really only set `if options.WebhookServer == nil {`

Reply:
And yeah this inheritance logic is very confusing to say the least, hence the deprecation :D